test/test_numpy.py

```diff
@@ -117,6 +117,156 @@ def test_numpy_array_d1_bool(self):
             b"[true,false,false,true]",
         )
 
+    def test_numpy_array_d1_datetime64_years(self):
+        self.assertEqual(
+            orjson.dumps(
+                numpy.array(
+                    [
+                        numpy.datetime64("2021"),
```

Comment:
It would help to generally test minimum year (and I don't know how that compares to datetime.datetime?), 3-digit-or-less (RFC 3339 requires they're zero padded to 4 digits), pre-epoch, something sort of current, and max.